### PR TITLE
feat(gsoc): add feature flag for vue simulator

### LINKED ISSUE #3768: Integration of cv-frontend-vue repository with the circuitverse (main) repository : part 1
### tasks to be done:

- [x] Add the cv-frontend-vue repository into main repository with the help of git submodules

- [x] setup the routes for the new simulator

- [x] update esbuild to configure the build of the vue simulator

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by Arnabdaz**
Fixes #3768 

#### Describe the changes you have made in this PR -

- [x] add vuesim feature flag for vue simulator
- [x] add option to enable / disable simulator from user profile settings
- [x] update simualtorcontroller to redirect to vue simualtor when feature flag is enabled 

### Screenshots of the changes (If any) -

N/A

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Review by vedant-jain03 — APPROVED** [review]
LGTM 
cc @VaibhavUpreti check for delpoyement compatability

**Comment by VaibhavUpreti** [comment]
Most of the things appear to be working correctly. I have a few questions @Arnabdaz :

1. Do we plan on merging #3797  before this PR? Without it, the simulator would not function and break the page for user, when viewing and launching(to edit) older circuits.

- I had tested these changes taking both PRs into account, then it works well.

2. Do developers need to clone the submodules when running the server? Is this the expected behaviour?

- If yes, we would have to update the documentation for cloning the submodule, installing all packages in `cv-frontend-vue` and then running bin/dev (probably can automate this later).

Else will give such error
```bash
22:39:51 js.1     | Error: package.json or package-lock.json is not found inside submodule directory
22:39:51 js.1     | error Command failed with exit code 1.
```

3. The project view modal seems to be broken, so if a user enables the vuesim then
- Multiple "view" button clicks trigger repeated page reconfirmation, 
- and the modal cannot load circuit data

broken

**Comment by Arnabdaz** [comment]
@VaibhavUpreti 

1. no that pr won't be merged as the api endpoints were decided to be changed and move in projects controller at the gsoc second phase that pr is already merged but not deployed #3861 

2. yes the devs are intended to clone the repo with either `--recursive` as `git clone https://github.com/CircuitVerse/CircuitVerse.git --recursive' 
or,
they can clone the repo normally and then use 
`git submodule update --init` to clone the cv-frontend-vue repo (this would be updated in the setup.md files also)
the error is intentionally added as in future vue simulator would be the main and only simulator 
we can update the error message and add the part `run git submodule update --init to fix this issue` or something like that.

3. the embedded circuit can't be loaded as the prs are not yet merged in cv-frontend-vue repo once they are merged it would work and other issues should also be fixed

**Review by VaibhavUpreti — APPROVED** [review]
LGTM, cc @tachyons

**Comment by vedant-jain03** [comment]
> Could you please update the documentation for this in a follow up PR? @Arnabdaz

For this we need to good documentation, we need some checkpoints:
- [x] for contributors
- [x] a perfect setup
- [x] contribution for simulator and cv(rails)
- [x] contribution for simulator only(vue)
- [x] an ideal PR flow

etc....

File: app/views/users/circuitverse/edit.html.erb [hunk 1]
```diff
@@ -42,6 +42,13 @@
               <div class="primary-checkpoint"></div>
             </label>
           </div>
+          <div class="field form-group users-edit-primary-checkpoint vuesim-checkbox">
+            <label for="vuesim" class="primary-checkpoint-container">
+              <h6>Enable New Simulator</h6>
+              <%= check_box_tag 'vuesim', '1', Flipper.enabled?(:vuesim, current_user), id: "vuesim" %>
+              <div class="primary-checkpoint"></div>
+            </label>
+          </div>
```
Comment (tachyons) [inline app/views/users/circuitverse/edit.html.erb:51]: Make this text i18n compatiable, add a warning and move to a separate PR

**Review by tachyons — CHANGES_REQUESTED** [review]

**Pull request closed without merging**